### PR TITLE
Mobile - Update editor initial HTML test for iOS and update Appium

### PR DESCRIPTION
**Pull request by geriux**
**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5197

## What?

It updates the initial HTML test for iOS using the `getBlockAtPosition` helper instead of manually adding retry logic when trying to locate the last block.

## Why?

There were quite a few failures lately related to the iOS test.

## How?

First, it updates the test to use `getBlockAtPosition` which has the latest code improvements to find a block in E2E tests.

It also updates:
- Appium to `1.22.3`
- iOS simulator to `iPhone 13`
- iOS to `15.4`

For Android:
- For CircleCI it updates the platform to `11.0`
- And the Appium version to: `1.22.1` (It doesn't support `1.22.3` yet)
- For the Github action, I couldn't get to update the current Android version `28`. I tried different approaches but it would fail to boot the emulator, it looks like the machine it uses it's too slow and it times out.

Dependabot updates that got merged before this PR:
- https://github.com/WordPress/gutenberg/pull/44748
- https://github.com/WordPress/gutenberg/pull/44744
- https://github.com/WordPress/gutenberg/pull/44743
- https://github.com/WordPress/gutenberg/pull/44741
- https://github.com/WordPress/gutenberg/pull/44749

## Testing Instructions
Circle CI should pass, maybe try restarting the jobs to see if they succeed again.

Gutenberg Mobile PRs should also pass, including full test checks.

## Screenshots or screencast 
N/A

### CODE REVIEW

File: packages/react-native-editor/__device-tests__/gutenberg-editor-initial-html.test.js [hunk 1]
```diff
@@ -16,22 +19,10 @@ describe( 'Gutenberg Editor Blocks test', () => {
 				lastBlockAccessibilityLabel
 			);
 		} else {
-			lastBlockElement = await editorPage.getLastElementByXPath(
-				lastBlockAccessibilityLabel
+			lastBlockElement = await editorPage.getBlockAtPosition(
```
Comment (fluiddot) [inline packages/react-native-editor/__device-tests__/gutenberg-editor-initial-html.test.js:22]: I noticed that `getBlockAtPosition` [has an option to auto-scroll to the element](https://github.com/WordPress/gutenberg/blob/fcaf478e3085ad7f0db4444de373ac919ae1f1dc/packages/react-native-editor/__device-tests__/pages/editor-page.js#L83), so we could use it also in the `if ( isAndroid() )` case to simplify the test, wdyt?

**Review by fluiddot — APPROVED** [review]
LGMT 🎊 ! Awesome work @geriux 🏅  ! 

After running the E2E tests three times all of them succeed ✅ , so looks like they're less flaky than before 🤞 .
[React Native E2E Tests (Android)](https://github.com/WordPress/gutenberg/actions/runs/3204193340)|[React Native E2E Tests (iOS)](https://github.com/WordPress/gutenberg/actions/runs/3204193372)
--|--
Screenshot 2022-10-07 at 14 33 57|Screenshot 2022-10-07 at 14 28 19

File: packages/react-native-editor/__device-tests__/gutenberg-editor-initial-html.test.js (hunk 1 above)
Comment (geriux) [reply packages/react-native-editor/__device-tests__/gutenberg-editor-initial-html.test.js:22]: Thanks for the suggestion! I tried it but it is not working for this test, it begins to scroll down but then it stops just returning:

`"//android.view.ViewGroup[contains(@content-desc, "Paragraph Block. Row 30")]" is still not visible after 25 retries!`

I'm going to merge this so we can see how it performs in trunk, if there's any tweaks needed I could investigate it further to unify it for both platforms.

Comment (fluiddot) [reply packages/react-native-editor/__device-tests__/gutenberg-editor-initial-html.test.js:22]: > Thanks for the suggestion! I tried it but it is not working for this test, it begins to scroll down but then it stops just returning:
> 
> "//android.view.ViewGroup[contains(@content-desc, "Paragraph Block. Row 30")]" is still not visible after 25 retries!

Ah, it's a shame that didn't work. Thanks for checking it 🙇 !

> I'm going to merge this so we can see how it performs in trunk, if there's any tweaks needed I could investigate it further to unify it for both platforms.

Sure, my comment was just a suggestion, we can explore in the future enhancements to this test case.

**Pull request merged**